### ISSUE TITLE
Put

### ISSUE THREAD

**Issue by joamatab** [gdsfactory/gdsfactory#1663]
Combine add_ref + connect into `put`

```python
c = gf.Component()

wg1 = c.add_ref(gf.components.waveguide(length=5))
wg2 = c.add_ref(gf.components.waveguide(length=10))
wg2.connect('o1', wg1['o2'])
```

equivalent

```python
c = gf.Component()

wg1 = c.put(gf.components.waveguide(length=5))
wg2 = c.put(gf.components.waveguide(length=10))
```

works well for 1x1 components,

for component with more than 1x1 ports

```python
c = gf.Component()
wg =  c.add_ref(gf.components.waveguide(length=5))
wg.chain(gf.components.waveguide(length=10)))
```

another option would be to add more functionality with `add_ref`

**Comment by thomasdorch** [comment]
This is a good idea. Maybe `put_all` might be more clear for components w/ more than 1x1 ports?

**Comment by tvt173** [comment]
I recommended the `chain` alternative over the `put` method. A few subtle but important reasons why...
- In the `put()` scheme, `put()` is a method of the parent component, rather than the previous reference to connect to. This implies that `Component` must keep track of a hidden state (the last component/port which was placed). I think this is a bit messy and error-prone... i.e. if we move lines around, suddenly our circuit will get jumbled up. It can also be a bit harder to follow at a glance.
- In the `ref.chain()` method, the previous reference is responsible for chaining the next reference... No new state needs to be tracked, and the connection is "correct by construction" since we are explicitly defining the two references to be connected.

In summary, I think we should prefer compositional methods vs prodecural when constructing circuits like this, and so I would prefer the `chain()` approach.

I think the original proposal was for 1x1 components, where it should be fairly straightforward to do this... If we want to allow n x m components, like @thomasdorch proposes, this gets a bit more complicated, because we need to figure out/specify which ports to connect, and we no longer have the expected return type of a single `ComponentReference`.

**Comment by lodoreds** [comment]
What about having default ports (e.g.: `o1` for input and `o2` for output, but could be changed by the user for electrical components) at which components are automatically put? Sequential placing of 1x1 components could then become very handy, while NxM components would still require the port to be specified in the usual way.

What do you think about the following syntax:
-  `add_ref(component).connect()` (both arguments to None): places components sequentially by joining the output port of the latest component with the default input port of the component to be added. In an empty cell, this could connect the first component to a default port in (0, 0);
- `add_ref(component).connect(ref.ports["o3"])` (only target port specified) connects the default input port of `component` to a selected port of `ref`. This could be useful when connecting a waveguide to a NxM component;
- `add_ref(component).connect("o5", ref.ports["o3"])`. This would be necessary for all other NxM components.

A possible implementation could use a global `current_port` variable that keeps track of the output port of the last component placed. This method could be very helpful for non-manhattan routing where many 1x1 components need to be placed back-to-back by hand.
